java/main/src/test/java/com/saucelabs/saucebindings/integration/PerformanceTest.java

```diff
@@ -0,0 +1,168 @@
+        SauceOptions sauceOptions = SauceOptions.chrome()
+                .setName(testName.getMethodName())
+                .setCapturePerformance()
+                .build();
+        session = new SauceSession(sauceOptions);
+        driver = session.start();
+    }
+
+    @Test
+    public void jankinessResults() {
+        driver.get("https://www.saucedemo.com");
+
+        JankinessResults jankinessResults = session.performance().getJankinessResults();
+        Assert.assertTrue(jankinessResults.getScore() > 0.7);
+    }
+
+    @Test
+    public void performanceResults() {
+        driver.get("https://www.saucedemo.com");
+
+        PerformanceResults performanceResults = session.performance().getResults();
+        // We don't actually care if this "passes" or "fails" just that these methods are returning correct values
+        if (performanceResults.isPassed()) {
+            Assert.assertTrue(performanceResults.getDetails().isEmpty());
+            Assert.assertNull(performanceResults.getReason());
+        } else {
+            Assert.assertTrue(performanceResults.getReason().contains("Metrics"));
+            Assert.assertFalse(performanceResults.getDetails().isEmpty());
+        }
+    }
+
+    @Test
+    public void performanceMetricResults() {
+        driver.get("https://www.saucedemo.com");
+
+        PerformanceResults performanceResults = session.performance().getResults("firstInteractive");
+        if (performanceResults.isPassed()) {
+            Assert.assertTrue(performanceResults.getDetails().isEmpty());
+            Assert.assertNull(performanceResults.getReason());
+        } else {
+            Map<String, Object> firstInteractive = (Map<String, Object>) performanceResults.getDetails().get("firstInteractive");
+            Assert.assertTrue((Long) firstInteractive.get("actual") < 5000);
+        }
+    }
+
+    @Test
+    public void performanceMetricsResults() {
+        driver.get("https://www.saucedemo.com");
+
+        List<String> metrics = new ArrayList<String>();
+        metrics.add("firstInteractive");
+        metrics.add("load");
+
+        PerformanceResults performanceResults = session.performance().getResults(metrics);
+        if (performanceResults.isPassed()) {
+            Assert.assertTrue(performanceResults.getDetails().isEmpty());
+            Assert.assertNull(performanceResults.getReason());
+        } else if (performanceResults.getDetails().get("firstInteractive") == null) {
+            Assert.assertNotNull(performanceResults.getDetails().get("load"));
+        } else {
+            Map<String, Object> firstInteractive = (Map<String, Object>) performanceResults.getDetails().get("firstInteractive");
+            Assert.assertTrue((Long) firstInteractive.get("actual") < 5000);
+        }
+    }
+
+    @Test
+    public void performanceDetails() {
+        driver.get("https://www.saucedemo.com");
+
+        PerformanceMetrics performanceMetrics = session.performance().getMetrics();
```

Comment:
This confuses me. I don't understand the difference between a `PerformanceMetrics` and `PerformanceResults`

Reply:
Maybe we rename PerformanceMetrics to PerformanceLog.

PerformanceResults uses Sauce Labs magic AI values. You can pass in specific metrics you care about and Sauce will pass/fail it based on whether we think it is in range.

If you want to assert on anything  based on a hard coded value instead of our magic, you need to first get the log.

I mean, maybe this should be:
```java
PerformanceLogs performanceLogs = session.logs().getPerformance();
```

Reply:
Makes sense. The API call would be what you said or `session.performance().getLogs()`

Reply:
The other part of this is that it isn't all the logs, just the logs for the last page transition. `session.logs()` should be getting all the logs for something, so maybe `session.performance().getLastLog()`?

Reply:
@titusfortner so if it's just for the last page transition then maybe?
`session.performance().getLastPageTransitionLog()`